addons/mail/static/tests/translation/translation_test.js

```diff
@@ -0,0 +1,50 @@
+/* @odoo-module */
+
+import { click, start, startServer, contains } from "@mail/../tests/helpers/test_utils";
+
+const spanishText = "Al mal tiempo, buena cara.";
+const englishText = "To bad weather, good face.";
+const rpc = { value: englishText, source: "Spanish", error: null };
+
+QUnit.module("Google Cloud Translation");
+
+QUnit.test("Toggle display of original/translated version of chatter message", async (assert) => {
+    let requestCount = 0;
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({});
+    pyEnv["mail.message"].create({
+        model: "res.partner",
+        body: spanishText,
+        author_id: pyEnv.odoobotId,
+        res_id: partnerId,
+    });
+    const { env, openFormView } = await start({
+        mockRPC(route) {
+            if (route === "/discuss/message/translate") {
+                requestCount++;
+                return rpc;
+            }
+        },
+    });
```

Comment:
Counters are not great in test, too imperative and require to understand everything to get what they intend to test.

Prefer `assert.step()` in the mocked RPC, and `assert.verifySteps()` in the assertions of test.

Reply:
Wonderful tool, thank you for pointing it out.